static/js/auth.js

```
@@ -364,138 +373,133 @@ export function createAuthModule(deps) {
+          await clearTokenState({ source: 'refresh_failed_after_401_in_verify' });
+          broadcastAuth(false, null, 'refresh_failed_after_401_in_verify');
           return false;
         }
       }
 
-      // For network errors or other issues, don't immediately clear auth state
-      // if we have cookies - the user might just have a temporary connection issue
-      const hasCookies = publicAuth.hasAuthCookies();
+      // For network errors (status 0 or no status) or other non-401/500 errors
+      const hasCookiesOnNetworkError = publicAuth.hasAuthCookies(); // Re-check, might have changed
       if (hasCookies && (error.status === 0 || !error.status)) {
```

ESLint

[error] 'hasCookies' is not defined.

(no-undef)